Labels in agent inspector are not visible

**Describe the bug**
Since recently the inspect view of an agent doesn't display the name of inspected variables:
![image](https://github.com/gama-platform/gama/assets/6374469/34f2b707-0f74-491f-90cb-314f45cef77a)
the left column should contain the names but it's empty.
The tooltip is work though:
![image](https://github.com/gama-platform/gama/assets/6374469/55574f84-9ac5-4994-80c6-32a3482cacde)

In the console I have this error stack, but it is right when I run the simulation, before opening the inspector view, I don't know if it's related. No new error message when I open the inspector:
```
!ENTRY org.eclipse.jface.text 4 0 2024-06-03 14:14:12.718
!MESSAGE java.lang.NullPointerException: Cannot invoke "String.replace(char, char)" because "label" is null
!STACK 0
java.util.concurrent.CompletionException: java.lang.NullPointerException: Cannot invoke "String.replace(char, char)" because "label" is null
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1770)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1760)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: java.lang.NullPointerException: Cannot invoke "String.replace(char, char)" because "label" is null
	at org.eclipse.ui.internal.editors.text.codemining.annotation.AnnotationCodeMining.sanitizeLabel(AnnotationCodeMining.java:57)
	at org.eclipse.ui.internal.editors.text.codemining.annotation.AnnotationCodeMining.<init>(AnnotationCodeMining.java:51)
	at org.eclipse.ui.internal.editors.text.codemining.annotation.AnnotationCodeMiningProvider.createCodeMining(AnnotationCodeMiningProvider.java:287)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
	at java.base/java.util.stream.SliceOps$1$1.accept(SliceOps.java:200)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
	at java.base/java.util.stream.SortedOps$RefSortingSink.end(SortedOps.java:400)
	at java.base/java.util.stream.Sink$ChainedReference.end(Sink.java:258)
	at java.base/java.util.stream.Sink$ChainedReference.end(Sink.java:258)
	at java.base/java.util.stream.Sink$ChainedReference.end(Sink.java:258)
	at java.base/java.util.stream.Sink$ChainedReference.end(Sink.java:258)
	at java.base/java.util.stream.Sink$ChainedReference.end(Sink.java:258)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:528)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575)
	at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260)
	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616)
	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:622)
	at java.base/java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:627)
	at org.eclipse.ui.internal.editors.text.codemining.annotation.AnnotationCodeMiningProvider.createCodeMinings(AnnotationCodeMiningProvider.java:274)
	at org.eclipse.ui.internal.editors.text.codemining.annotation.AnnotationCodeMiningProvider.lambda$0(AnnotationCodeMiningProvider.java:224)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
	... 6 more
Key released: 	
```

when closing the simulation with this view still open I get this message, which may or may not be linked but it also seem like a (minor) bug:
```
An internal error has occurred.
Cannot invoke "gama.core.kernel.simulation.SimulationClock.getCycle()" because the return value of "gama.core.runtime.IScope.getClock()" is null
```

**To Reproduce**
Steps to reproduce the behavior:
1. open any model with agents
2. open the inspector on one agent of the simulation
4. See error

**Expected behavior**
We see the labels as it was the case before

--- a/gama.core/src/gama/gaml/variables/Variable.java
+++ b/gama.core/src/gama/gaml/variables/Variable.java
@@ -512,9 +512,9 @@ public class Variable extends Symbol implements IVariable {
 	public Variable(final IDescription sd) {
 		super(sd);
 		final VariableDescription desc = (VariableDescription) sd;
-		setName(sd.getName());
-		parameter = null;
-		category = null;
+		setName(sd.getName());		
+		parameter = desc.getParameterName();
+		category = getLiteral(IKeyword.CATEGORY, null);
 		updateExpression = getFacet(IKeyword.UPDATE);
 		functionExpression = getFacet(IKeyword.FUNCTION);
 		initExpression = getFacet(IKeyword.INIT);
@@ -522,13 +522,8 @@ public class Variable extends Symbol implements IVariable {
 		onChangeExpression = getFacet(IKeyword.ON_CHANGE);
 		isNotModifiable = desc.isNotModifiable();
 		type = desc.getGamlType();
-		// computeSpeciesConst();
 	}
 
-	// private void computeSpeciesConst() {
-	// isSpeciesConst = isNotModifiable && updateExpression == null && functionExpression == null && getter == null
-	// && setter == null && (initExpression == null || initExpression.isConst());
-	// }
 
 	/**
 	 * Builds the helpers.
@@ -557,9 +552,7 @@ public class Variable extends Symbol implements IVariable {
 
 		// }
 
-		// VariableDescription var = (VariableDescription) description;
 		SpeciesDescription sp = species.getDescription();
-		// if (var.isBuiltIn()) return;
 		Class base = sp.getJavaBase();
 		if (base == null) return;
 		List<GamaHelper> helpers = new ArrayList<>();

--- a/gama.ui.shared/src/gama/ui/shared/parameters/AbstractEditor.java
+++ b/gama.ui.shared/src/gama/ui/shared/parameters/AbstractEditor.java
@@ -104,12 +104,14 @@ public abstract class AbstractEditor<T> implements SelectionListener, ModifyList
 	protected EditorsGroup parent;
 
 	/** The editor toolbar. */
+	@SuppressWarnings("rawtypes")
 	protected EditorToolbar editorToolbar;
 
 	/** The editor label. */
 	protected EditorLabel editorLabel;
 
 	/** The editor control. */
+	@SuppressWarnings("rawtypes")
 	protected EditorControl editorControl;
 
 	/**
@@ -124,7 +126,6 @@ public abstract class AbstractEditor<T> implements SelectionListener, ModifyList
 	 * @param l
 	 *            the l
 	 */
-	@SuppressWarnings ("unchecked")
 	public AbstractEditor(@Nonnull final IAgent a, @Nonnull final IParameter parameter,
 			@Nullable final EditorListener<T> l) {
 		param = parameter;
@@ -357,7 +358,7 @@ public abstract class AbstractEditor<T> implements SelectionListener, ModifyList
 	 *
 	 * @return the editor toolbar
 	 */
-	@SuppressWarnings ("unchecked")
+	@SuppressWarnings ({ "unchecked", "rawtypes" })
 	EditorToolbar createEditorToolbar() {
 		editorToolbar = new EditorToolbar(this, parent);
 		updateToolbar();
@@ -369,6 +370,7 @@ public abstract class AbstractEditor<T> implements SelectionListener, ModifyList
 	 *
 	 * @return the editor control
 	 */
+	@SuppressWarnings("rawtypes")
 	EditorControl createEditorControl() {
 		boolean isCombo = param.getAmongValue(getScope()) != null;
 		boolean isEditable = param.isEditable();

